spec/contract.spec.json

```diff
@@ -0,0 +1,105 @@
+{
+  "id": "truffle-contract-schema.json",
+  "$schema": "http://json-schema.org/schema#",
+  "type": "object",
+  "properties": {
+    "contractName": { "$ref": "#/definitions/ContractName" },
+    "abi": { "$ref": "#/definitions/ABI" },
+    "bytecode": { "allOf": [
+      { "$ref": "#/definitions/Bytecode" },
+      { "description": "Bytecode sent as contract-creation transaction data" }
+    ]},
+    "deployedBytecode": { "allOf": [
+      { "$ref": "#/definitions/Bytecode" },
+      { "description": "On-chain deployed contract bytecode" }
+    ]},
+    "sourceMap": { "allOf": [
+      { "$ref": "#/definitions/SourceMap" },
+      { "description": "Source mapping for contract-creation transaction data bytecode" }
+    ]},
+    "deployedSourceMap": { "allOf": [
+      { "$ref": "#/definitions/SourceMap" },
+      { "description": "Source mapping for contract bytecode" }
+    ]},
+    "source": { "$ref": "#/definitions/Source" },
+    "sourcePath": { "$ref": "#/definitions/SourcePath" },
+    "ast": { "$ref": "#/definitions/AST" },
+    "address": { "$ref": "#/definitions/Address" },
+    "networks": {
+      "patternProperties": {
+        "^[a-zA-Z0-9]+$": { "$ref": "#/definitions/Network" }
+      },
+      "additionalProperties": false
+    },
+    "schemaVersion": { "$ref": "#/definitions/SchemaVersion" },
+    "updatedAt": {
+      "type": "string",
+      "format": "date-time"
+    }
+  },
+  "additionalProperties": false,
+  "required": [
+  ],
+  "definitions": {
+    "ContractName": {
+      "type": "string",
+      "pattern": "^[a-zA-Z_][a-zA-Z0-9_]*$"
+    },
+    "Address": {
+      "type": "string",
+      "pattern": "^0x[a-fA-F0-9]{40}$"
+    },
+    "ABI": {
+      "type": "array"
+    },
+    "Bytecode": {
+      "type": "string",
+      "pattern": "^0x0$|^0x([a-fA-F0-9]{2}|__[a-zA-Z0-9_]{38})+$"
+    },
+    "Source": {
+      "type": "string"
+    },
+    "SourceMap": {
+      "type": "string"
+    },
+    "SourcePath": {
+      "type": "string"
+    },
+    "AST": {
+      "type": "object"
```

Comment:
Definitely not in scope - at least, don't write your own validator for it. I'd be surprised if solidity doesn't already have one. If they do, treat it like the abi above; if they don't, leave it alone for now.